app/controllers/search_controller.rb

```diff
@@ -82,6 +82,14 @@ def users
     render json: { result: user_search }
   end
 
+  def usernames
+    usernames = Search::User.search_usernames(params[:username])
+
+    render json: { result: usernames }
+  rescue Search::Errors::Transport::BadRequest
```

Comment:
Out of curiosity, what happens if someone hits this endpoint a ton of times? I'm thinking of the frontend implementation of this and how it will need to throttle requests to this endpoint so that we're not searching the entire DB of usernames more times than necessary, but just wondering what kind of error would be raised if that were to happen? 

(I am an ES newbie 😄)

Reply:
That is a great question! Ideally, we don't want someone to spam this endpoint bc it would slow down ES which is why we have throttling on it!
```ruby
module Rack
  class Attack
    throttle("search_throttle", limit: 5, period: 1) do |request|
      if request.path.starts_with?("/search/")
        track_and_return_ip(request.env["HTTP_FASTLY_CLIENT_IP"])
      end
    end
```

Reply:
Ah, good to know! I didn't realize we had already accounted for this in `Rack::Attack`, but makes total sense 👍

Reply:
Your question made me check the permissions on this endpoint as well. Given you cannot mention someone without being a user I put it behind our authentication which further decreases the possibility of abuse. This is why questions are so valuable in PRs!!!!!!

Reply:
AH! I should have remember to check for that myself during review! That seems good + important to do from a security perspective, too.